packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DocumentController.java

```diff
@@ -362,16 +369,57 @@ private ResponseEntity<Void> uploadDocumentHelper(
 		try {
 			// upload file to S3
 			final Integer status = documentAssetService.uploadFile(documentId, fileName, fileEntity);
+			final Optional<DocumentAsset> document = documentAssetService.getAsset(documentId, permission);
+
+			if (document.isPresent()) {
+				try {
+					// Convert BufferedImage to byte[]
+					PDDocument pdfDocument = Loader.loadPDF(fileEntity.getContent().readAllBytes());
+					PDFRenderer pdfRenderer = new PDFRenderer(pdfDocument);
+					BufferedImage firstPageImage = pdfRenderer.renderImageWithDPI(0, 100);
+
+					int height = firstPageImage.getHeight();
+					int width = firstPageImage.getWidth();
+					int topHalfHeight = height / 2;
+
+					// Create a sub-image (top half)
+					BufferedImage topHalfImage = firstPageImage.getSubimage(0, 0, width, topHalfHeight);
+
+					// Resize the image to a width of 225px while maintaining aspect ratio
+					int newWidth = 225;
+					int newHeight = (newWidth * topHalfHeight) / width; // Maintain the aspect ratio
+
+					// Create a new BufferedImage for the resized image
+					Image resizedImage = topHalfImage.getScaledInstance(newWidth, newHeight, Image.SCALE_SMOOTH);
+					BufferedImage resizedBufferedImage = new BufferedImage(newWidth, newHeight, BufferedImage.TYPE_INT_ARGB);
+
+					Graphics2D g2d = resizedBufferedImage.createGraphics();
+					g2d.drawImage(resizedImage, 0, 0, null);
+					g2d.dispose();
+
+					ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+					ImageIO.write(resizedBufferedImage, "png", outputStream);
+
+					byte[] thumbnailBytes = outputStream.toByteArray();
+					outputStream.close();
+
+					document.get().setThumbnail(thumbnailBytes);
+					pdfDocument.close();
```

Comment:
There are a lot of calls to `close` various things in here that won't get triggered if there is an exception at some point in creating the thumbnail... consider using the `finally` block for these? Somethings will have to be re-scoped for this to work

```
try{
...
} catch (Exception e) {
...
} finally {
g2d.dispose()
outputStream.close()
...
}
```